### ISSUE TITLE
Add an option to remember user decision of inclusion or exclusion of cross-referenced entries when copying

### ISSUE THREAD

**Issue by HoussemNasri** [JabRef/jabref#12341]
Depends on #12340 

## What?
As part of implementing the include/exclude cross-references option in (#12340) when copying entries, we need to give users the option to preserve their selection across sessions for handling cross-references during the copy-paste operation.

## How?
Add a "Remember my decision" checkbox to the "Copy to" dialog. When checked, this option will apply the user’s current selection (whether to include or exclude cross-referenced entries) automatically in future copy operations, without displaying the dialog again.

Store user decision in `JabRefGuiPreferences`.

## Definition of Done
- Unit tests to verify that the user's preference for cross-references is remembered and applied correctly.

**Comment by yashpandey06** [comment]
@HoussemNasri ...that seems interesting can i work on this ?

**Comment by yashpandey06** [comment]
@HoussemNasri just a curious question ...are we planning to take a part in the upcoming GSOC session ?

**Comment by HoussemNasri** [comment]
> that seems interesting can i work on this ?

This is a task to implement a complete new feature, unfortunately the tasks are not independent, #12339 is the first task in the list. You can ask to be assigned there if the task looks interesting.

**Comment by HoussemNasri** [comment]
> just a curious question ...are we planning to take a part in the upcoming GSOC session ?

Who are we? 😅

**Comment by yashpandey06** [comment]
> > just a curious question ...are we planning to take a part in the upcoming GSOC session ?
> 
> Who are we? 😅

Apologies for the confusion...I mean the org/project

**Comment by HoussemNasri** [comment]
We probably will, there is not yet any official information from Google, but we expect that we will need to finalize our ideas at the beginning of the year.

**Comment by priyanshu16095** [comment]
How can I assign this task to myself?

**Comment by priyanshu16095** [comment]
I am interested in working on this.

**Comment by priyanshu16095** [comment]
![Image](https://github.com/user-attachments/assets/b3849a81-8e78-415e-8921-87a7ac71c7f2)
Is the UI okay according to the requirements given in the issue?

**Comment by priyanshu16095** [comment]
Do I also need to write tests for the Preference class? Other Preference classes do not have any tests written.

**Comment by Siedlerchr** [comment]
No, we focus on testing the important stuff, mostly the logic stuff. We do have some UI tests, but they are not easy. 
However, probably you can also stuff like viewModels. They just expose their data through bindings.

**Comment by priyanshu16095** [comment]
I have created a draft PR. Please review it and provide any suggestions.

**Comment by HoussemNasri** [comment]
Closed by https://github.com/JabRef/jabref/pull/12374
